### PR TITLE
MCM-15: CRUD MedicalEntity

### PR DESCRIPTION
**Pull request by taniarodriguez**
CRUD MedicalEntity entity
Test for this entity

### CODE REVIEW

File: src/entities/IPS.entity.ts [hunk 1]
```diff
@@ -16,14 +16,13 @@
...
 import { Length, IsOptional, IsUUID, IsDefined, IsBoolean} from 'class-validator';
-import { Event } from './event.entity';
 import { TypeDocument } from './typeDocument.entity';
 
 
 @Entity()
-export class MedicalCenter{ 
+export class IPS{ 
 
     @PrimaryGeneratedColumn("uuid")
```
Comment (EngAndres) [inline src/entities/IPS.entity.ts:27]: defined or optional?

File: src/entities/IPS.entity.ts [hunk 2]
```diff
@@ -55,6 +54,7 @@ export class MedicalCenter{
     @IsBoolean()
```
Comment (EngAndres) [inline src/entities/IPS.entity.ts:54]: defined or optional?

File: src/entities/IPS.entity.ts [hunk 3]
```diff
@@ -55,6 +54,7 @@ export class MedicalCenter{
...
     isActive: boolean;
 
-    @ManyToOne (() => TypeDocument, (typeDoocument) => typeDoocument.medicals)
+    @ManyToOne (() => TypeDocument, (typeDocument: TypeDocument) => typeDocument.ips)
+    @IsDefined()
```
Comment (EngAndres) [inline src/entities/IPS.entity.ts]: here this tag is unnecesary

File: src/entities/medicalEntity.entity.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    @Length(0,30)
+    numberDocument: string;
+    
+    @Column ({nullable: false})
+    @IsDefined()
+    @IsBoolean()
+    isActive: boolean;
+    
+    @ManyToOne (() => TypeDocument, (typeDocument: TypeDocument) => typeDocument.medicalEntity)
+    @IsDefined()
```
Comment (EngAndres) [inline src/entities/medicalEntity.entity.ts]: this tag is unnecesary.

**Review by EngAndres — CHANGES_REQUESTED** [review]
Review comments and apply all changes you consider are necessary.

File: src/entities/IPS.entity.ts (hunk 3 above)
Comment (taniarodriguez) [reply src/entities/IPS.entity.ts]: Ok

File: src/entities/medicalEntity.entity.ts (hunk 4 above)
Comment (taniarodriguez) [reply src/entities/medicalEntity.entity.ts]: Ok

File: src/entities/IPS.entity.ts (hunk 2 above)
Comment (taniarodriguez) [reply src/entities/IPS.entity.ts:54]: This field IsDefined.
 @Column ({nullable: false})
    @IsDefined()
    @IsBoolean()
    isActive: boolean;

When we create a new register, the field IsActive is set here
medicalData.isActive = true;
And when the register is updated, it check if the field exist
Utilities.checkParameterExistence(medicalData.isActive);

File: src/entities/IPS.entity.ts (hunk 1 above)
Comment (taniarodriguez) [reply src/entities/IPS.entity.ts:27]: The field is Optional

@PrimaryGeneratedColumn("uuid")
    @IsUUID()
    @IsOptional()
    id: string;

File: src/entities/IPS.entity.ts (hunk 2 above)
Comment (EngAndres) [reply src/entities/IPS.entity.ts:54]: ok, it is right

**Review by EngAndres — APPROVED** [review]

**Pull request merged**